lib/OpenQA/WebAPI/Controller/Running.pm

```diff
@@ -127,10 +121,9 @@ sub streamtext ($self, $file_name, $start_hook = undef, $close_hook = undef) {
                 $size = -s $logfile;
             }
 
-            # Zero tolerance for any shenanigans with the logfile, such as
-            # truncation, rotation, etc.
             my @st = stat $logfile;
-            return $close_connection->() unless @st && $st[1] == $ino && $st[3] > 0 && $st[7] >= $size;
+            die "Zero tolerance for any shenanigans with the logfile, such as truncation, rotation, etc."
```

Comment:
We should likely close the connection here as the code previously did. Otherwise we leave the client waiting for further data until a timeout occurs. Placing a `die` here seems generally wrong as it would likely just be caught by Mojolicious' handler. The message has also not enough context to be useful in the server-side logs (where it would end up). If we wanted better error handling, then an error should likely go to the client in form of a bad return code.

Reply:
ok, I agree. All CI tests passing show that we are missing test coverage here